src/App.js

```diff
@@ -16,16 +16,24 @@ function App() {
    return setContacts(displayedContacts)
   }
   const sortByName = () => {
-    const sortedContacts = [...contacts].sort((a, b) => { return (a > b ? 1 : (a === b ? 0 : -1)) })
-    setContacts(sortedContacts);
+    const sortedByName = [...contacts.sort((a, b) => { return (a.name > b.name ? 1 : (a.name === b.name ? 0 : -1)) })]
+    return setContacts(sortedByName);
   }
+  const sortByPopularity = () => {
+    const sortedByPopularity = [...contacts.sort((a, b) => { return (a.popularity > b.popularity ? -1 : (a.popularity === b.popularity ? 1 : 0)) })]
+    return setContacts(sortedByPopularity);
+  }
```

Comment:
You should sort copied array, so have square brackets only around `...contacts`, and not the whole expression

Alternative solution:

```
  const sortedByPopularity = () => {
    setContacts(contacts => [...contacts].sort((a,b) => b.popularity - a.popularity))
  }
```

Reply:
it makes sense :)